dotnet/src/Svg.cs

```diff
@@ -0,0 +1,162 @@
+﻿using Microsoft.AspNetCore.Html;
+using System;
+using System.Collections.Concurrent;
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+
+namespace StackExchange.StacksIcons;
+
+public static partial class Svg
+{
+    private const int MaxReasonableSize = 4500;
+    public static readonly SvgImage Empty = new(string.Empty);
+
+    public static partial class Spot
+    {
+        /// <inheritdoc cref="Svg.GetImage(string?, bool, bool)"/>
+        internal static SvgImage GetImage([CallerMemberName] string? fileName = null) =>
+            // For spots, we intentionally want to bypass the size check on these rarely used items to make maintenance easier.
+            Svg.GetImage(fileName, isSpot: true, bypassSizeCheck: true);
+    }
+
+    /// <summary>
+    /// A hash that changes when any of the icons change or one gets added, removed, etc. Calculated in GetImage() by replacing this
+    /// value with a hash of the old value combined with the file contents. Thus relies on the fact that C# guarantees in-order execution
+    /// of static initializers (because the files have to be in the same order every time). The JS uses this value as a cache breaker when
+    /// downloading any SVG icon.
+    /// </summary>
+    public static string CombinedCacheBreaker { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// Gets an <see cref="SvgImage"/> for caching and reuse.
+    /// </summary>
+    /// <param name="fileName">The filename to grab, defaults to the caller's name via <see cref="CallerMemberNameAttribute"/>.</param>
+    /// <param name="isSpot">Whether the image is a Spot or not.</param>
+    /// <param name="bypassSizeCheck">Whether to allow bypassing the size check, for things we know not to repeat in a page (stil shouldn't be huge).</param>
+    /// <returns>The <see cref="SvgImage"/> to cache.</returns>
+    internal static SvgImage GetImage([CallerMemberName] string? fileName = null, bool isSpot = false, bool bypassSizeCheck = false)
+    {
+        if (string.IsNullOrEmpty(fileName))
+        {
+            return Empty;
+        }
+
+        try
+        {
+            var imageString = Helpers.GetSvg(fileName!, isSpot);
+#if DEBUG
+            // Okay so now we have the svg asset loaded into the string,
+            // if we're on DEBUG let's do some gut checking on the .svg itself.
+            // This way we throw in dev.
+            if (imageString is null)
+            {
+                throw new Exception($"Unable to find {fileName} in embedded resources.");
+            }
+
+            if (imageString.Length > MaxReasonableSize && !bypassSizeCheck)
```

Comment:
Agreed 1000%. When I ported the code from downstream to here, I had this thought myself. For now, I'm going to leave this in place since it only runs in DEBUG mode.